.github/workflows/build.yml

```diff
@@ -1,39 +1,111 @@
+
+on: [push]
 
 jobs:
-  build:
+  install:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install and build
+        run: |
+          yarn
+          yarn build
+
+      - name: Save build folder
+        uses: actions/upload-artifact@v3
+        with:
+          name: build
+          if-no-files-found: error
+          path: dist
+
+  contracts-node-binary:
     runs-on: ubuntu-latest
+    steps:
+      - run: |
+          curl -L "https://github.com/paritytech/substrate-contracts-node/releases/download/v0.15.1/substrate-contracts-node-linux.tar.gz" -O
+          ls -lash
+      - name: Save node artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: substrate-contracts-node
+          if-no-files-found: error
+          path: substrate-contracts-node-linux.tar.gz
 
+  ui-chrome-tests:
+    timeout-minutes: 15
+    runs-on: ubuntu-latest
+    container:
+      image: cypress/browsers:node14.15.0-chrome96-ff94
+      options: --user 1001
+    needs: [install, contracts-node-binary]
     strategy:
+      # when one test fails, DO NOT cancel the other
+      # containers, because this will kill Cypress processes
+      # leaving the Dashboard hanging ...
+      # https://github.com/cypress-io/github-action/issues/48
+      fail-fast: false
       matrix:
-        node-version: [14.x]
-
+        # run copies of the current job in parallel
+        containers: [1, 2]
     steps:
-      - uses: actions/checkout@v2
-      - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v2
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Prepare test environment
+        uses: ./.github/actions/prepare
+
+      - name: 'UI Tests - Chrome'
+        uses: cypress-io/github-action@v2
         with:
-          node-version: ${{ matrix.node-version }}
-      - name: Cache Node.js modules
-        uses: actions/cache@v2
-        id: yarn-cache
+          start: yarn start
+          wait-on: 'http://localhost:8081'
+          wait-on-timeout: 120
+          browser: chrome
+          record: true
+          parallel: true
+          group: 'UI - Chrome'
+        env:
+          CYPRESS_PROJECT_ID: ${{ secrets.CYPRESS_PROJECT_ID }}
+          CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  ui-firefox-tests:
+    timeout-minutes: 15
+    runs-on: ubuntu-latest
+    container:
+      image: cypress/browsers:node14.15.0-chrome96-ff94
+      options: --user 1001
+    needs: [install, contracts-node-binary]
+    strategy:
+      # when one test fails, DO NOT cancel the other
+      # containers, because this will kill Cypress processes
+      # leaving the Dashboard hanging ...
+      # https://github.com/cypress-io/github-action/issues/48
+      fail-fast: false
+      matrix:
+        # run copies of the current job in parallel
+        containers: [1, 2]
```

Comment:
Looks great! Curious why two containers are needed?

Reply:
We can specify up to 5 containers here. I added 2 because we only have 2 specs at the moment and this way they run in parallel
https://docs.cypress.io/guides/guides/parallelization